browser/components/RadzChart.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+
+const d3 = require('d3');
+
+interface Props {
+  data: object[];
+  centerText: string;
+  height: number;
+  width: number;
+  id: string;
+  cornerRadius: number;
+  padAngle: number;
+  centerTextSize: string;
+  centerTextdx: string;
+}
+
+export default class Rad extends React.Component<Partial<Props>, {}> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: [],
+    };
+  }
+
+  componentDidMount() {
+    const dataset = this.props.data;
+    this.generateChart(dataset);
+  }
+
+  componentWillReceiveProps() {
+    this.generateChart(this.props.data);
+  }
+
+  generateChart(data) {
+    const width = this.props.width;
+    const height = this.props.height;
+    const radius = Math.min(width, height) / 2;
+
+    function midAngle(d) {
+      return d.startAngle + (d.endAngle - d.startAngle) / 2;
+    }
+
+    const color = d3.scaleOrdinal(d3.schemeCategory10);
+
+    const pie = d3
+      .pie()
+      .value(function(d) {
+        return d.value;
+      })
+      .sort(null);
+
+    const arc = d3
+      .arc()
+      .outerRadius(radius * 0.8)
+      .innerRadius(radius * 0.65)
+      .cornerRadius(this.props.cornerRadius)
+      .padAngle(this.props.padAngle);
+
+    const outerArc = d3
+      .arc()
+      .outerRadius(radius * 0.9)
+      .innerRadius(radius * 0.9);
+
+    // remove previous svg each time from the div
+    d3.select('#' + this.props.id).remove();
+
+    const svg = d3
+      .select('#radzChart' + this.props.id)
+      .append('svg')
+      .attr('width', width)
+      .attr('height', height)
+      .attr('id', this.props.id)
+      .append('g')
+      .attr('transform', 'translate(' + width / 2 + ',' + height / 2 + ')');
+
+    const label = svg
+      .select('.labelName')
+      .selectAll('text')
+      .data(data)
+      .enter()
+      .append('text')
+      .attr('dy', '.35em')
+      .html(function(d) {
+        return d.data[label] + ': <tspan>' + d.value + '</tspan>';
+      })
+      .attr('class', 'labelName')
+      .attr('transform', function(d) {
+        const pos = outerArc.centroid(d);
+        pos[0] = radius * 0.95 * (midAngle(d) < Math.PI ? 1 : -1);
+        return 'translate(' + pos + ')';
+      })
+      .style('text-anchor', function(d) {
+        return midAngle(d) < Math.PI ? 'start' : 'end';
+      });
+
+    const g = svg
```

Comment:
This is not descriptive of what the parameter contains. Please fix.

Reply:
g is a sub-element of <svg>. its pretty cool, you can apply same transformation all the elements inside <g> tag. here's more: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/g

Reply:
A comment here might be a good compromise, since `g` is the actual SVG tag name for a group.